src/Microsoft.DotNet.VersionTools/Automation/DependencyUpdateUtils.cs

```diff
@@ -11,34 +12,53 @@ namespace Microsoft.DotNet.VersionTools.Automation
     public static class DependencyUpdateUtils
     {
         /// <summary>
-        /// Runs the updaters given using buildInfo sources, and returns information about the
-        /// updates performed, such as a recommended commit name based on the BuildInfos used.
+        /// Runs the updaters given using given dependency info sources. Returns information about
+        /// the updates performed, such as a recommended commit name.
         /// </summary>
         public static DependencyUpdateResults Update(
             IEnumerable<IDependencyUpdater> updaters,
-            IEnumerable<DependencyBuildInfo> buildInfoDependencies)
+            IEnumerable<IDependencyInfo> externalDependencyInfos,
+            Action<IDependencyInfo> modifyUpdaterDependencyInfo = null)
         {
-            IEnumerable<BuildInfo> distinctUsedBuildInfos = GetUpdateTasks(updaters, buildInfoDependencies)
+            IEnumerable<DependencyUpdateTask> updateTasks = GetUpdateTasks(
+                updaters,
+                externalDependencyInfos,
+                modifyUpdaterDependencyInfo);
+
+            IDependencyInfo[] distinctUsedDependencyInfos = updateTasks
                 .Select(task =>
                 {
                     task.Start();
                     return task.Result;
                 })
-                .SelectMany(results => results.UsedBuildInfos)
+                .SelectMany(results => results.UsedInfos)
                 .Distinct()
                 .ToArray();
 
-            return new DependencyUpdateResults(distinctUsedBuildInfos);
+            return new DependencyUpdateResults(distinctUsedDependencyInfos);
         }
 
         /// <summary>
         /// Returns the tasks necessary to update dependencies.
         /// </summary>
         public static IEnumerable<DependencyUpdateTask> GetUpdateTasks(
             IEnumerable<IDependencyUpdater> updaters,
-            IEnumerable<DependencyBuildInfo> buildInfoDependencies)
+            IEnumerable<IDependencyInfo> externalDependencyInfos,
+            Action<IDependencyInfo> modifyUpdaterDependencyInfo = null)
         {
-            return updaters.SelectMany(updater => updater.GetUpdateTasks(buildInfoDependencies));
+            IDependencyInfo[] updaterDependencyInfos = updaters
+                .OfType<IDependencyInfoProvider>()
+                .SelectMany(updater => updater.CreateDependencyInfos())
+                .ToArray();
+
+            foreach (var info in updaterDependencyInfos)
+            {
+                modifyUpdaterDependencyInfo?.Invoke(info);
```

Comment:
This is a case where I think having the caller create the DependencyInfos would help.  The caller can manipulate the DependencyInfo anyway it wants, and we don't need this `modifyUpdaterDependencyInfo` action at all here.